crates/control/src/role_assignment.rs

```diff
@@ -97,9 +98,9 @@ impl RoleAssignment {
         {
             role = match context.player_number {
                 PlayerNumber::One => Role::Keeper,
-                PlayerNumber::Two => Role::DefenderRight,
-                PlayerNumber::Three => Role::StrikerSupporter,
-                PlayerNumber::Four => Role::DefenderLeft,
+                PlayerNumber::Two => *context.optional_roles.get(0).unwrap_or(&Role::default()),
```

Comment:
What do you think of something like this:

```suggestion
                PlayerNumber::Two => context
                    .optional_roles
                    .get(0)
                    .copied()
                    .unwrap_or_else(|| Role::default()),
```

Reply:
done

Reply:
Clippy gave me:  `.unwrap_or_else(Role::default),`

done